src/v/kafka/server/tests/alter_config_test.cc

```diff
@@ -1368,6 +1373,82 @@ FIXTURE_TEST(test_unlicensed_alter_configs, alter_config_test_fixture) {
           failure);
     }
 
+    // Specific tests for schema id validation subject name strategy
+    {
+        using sns = pandaproxy::schema_registry::subject_name_strategy;
+
+        const auto full_validation = props_t{
+          with(kafka::topic_property_record_key_schema_id_validation, true),
+          with(kafka::topic_property_record_value_schema_id_validation, true),
+        };
+        test_cases.emplace_back(
+          "set_key_sns",
+          full_validation,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_key_subject_name_strategy,
+              sns::topic_name)},
+          },
+          failure);
+        test_cases.emplace_back(
+          "set_value_sns",
+          full_validation,
+          alter_props_t{
+            {set(
+              kafka::topic_property_record_value_subject_name_strategy,
+              sns::topic_name)},
```

Comment:
`topic_name` is the default; I'm wondering if explicitly setting this to the default should be considered a change? I don't think it's a problem.

Reply:
>setting this to the default should be considered a change?

I think so. Since it only "counts" if schema id verification is enabled on the topic, seems reasonable to restrict any alter config that changes its behavior. "default SNS and ON" is still a license violation.

If there are strong opinions I could be easily persuaded to go the other way though.